awssql/utils/sliding_expiration_cache.go

```diff
@@ -74,6 +74,8 @@ func (c *SlidingExpirationCache[T]) Put(key string, value T, itemExpiration time
 }
 
 func (c *SlidingExpirationCache[T]) Get(key string, itemExpiration time.Duration) (T, bool) {
+	c.cleanupIfExpired(key)
```

Comment:
I think we can remove the reflect on this line: https://github.com/aws/aws-advanced-go-wrapper/blob/8748b9f201898a2da318fbdc197172cd77d1cab0/awssql/utils/sliding_expiration_cache.go#L99-L101.